editor/editor_node.cpp

```diff
@@ -2373,6 +2373,19 @@ void EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) {
 			ScriptEditor::get_singleton()->get_debugger()->set_live_debugging(!ischecked);
 			EditorSettings::get_singleton()->set_project_metadata("debug_options", "run_live_debug", !ischecked);
 
+		} break;
+		case RUN_LISTEN_DEBUG: {
```

Comment:
From reading the code I'm not sure I understand how this branch will be "listening" to new device connections? The only difference with `RUN_LIVE_DEBUG` seems to be that it stops the debugger, and it does not set the `run_live_debug` metadata.

Reply:
It starts the debugger when the checkbox is not checked and when you toggle the checkbox off it stops the debugger.

Directly calling `ScriptEditor::get_singleton()->get_debugger()->start();` starts the debugger but through testing there seems to be a bug regarding resetting the debugger state when the remote device has disconnected. If the remote device has never connected, it will wait correctly, not after it disconnects.